preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/bdtopo/BDTopoGISLayers.groovy

```diff
@@ -112,6 +112,32 @@ IProcess importPreprocess(){
             def tmp_imperv_surface_activite = 'TMP_IMPERV_SURFACE_ACTIVITE_' + uuid
             def input_impervious = 'INPUT_IMPERVIOUS'
 
+            // -------------------------------------------------------------------------------
+            // Control the SRIDs from input tables
+
+            def list = [tableBuildIndifName, tableIrisName, tableBuildIndusName, tableBuildRemarqName,
+                        tableRoadName, tableRailName, tableHydroName, tableVegetName,
+                        tableImperviousSportName, tableImperviousBuildSurfName,
+                        tableImperviousRoadSurfName, tableImperviousActivSurfName]
+
+            // The SRID is stored and initialized to -1
+            def srid = -1
+
+            // For each tables in the list, we check the SRID and compare to the srid variable. If different, the process is stopped
+            for(String name : list){
+                def table = datasource.getTable(name)
```

Comment:
Should be more efficient to query the geometry_columns view with the list of tables
sql.row(select srid from geometry_columns where F_TABLE_NAME in('listoftables'));

Reply:
To be more clear (@ebocher don't forget to use markdown syntax to embed code), the previous groovy instruction return an `List` of `GroovyResultSet`. once you have the list, you can request it to get all the SRIDs.

@ebocher with your instruction, we can't get the name of the problematic table. Will the usage of SQL request does a by gain of time as this code section will be done once ?